src/cpp/session/modules/rmarkdown/NotebookQueue.cpp

```diff
@@ -283,7 +283,7 @@ class NotebookQueue : boost::noncopyable
          execContext_->onExprComplete();
          
       ExecRange range;
-      std::string code = execUnit_->popExecRange(&range, mode);
+      std::string code = execUnit_->popExecRange(&range, mode, execContext_->engine());
```

Comment:
passing the engine in as a param here so we don't have to parse it out from the code itself